### PR TITLE
DIV-3952 Co-Respondent Linking

### PR DESCRIPTION
**Pull request by alexwakeman**
Adds handling for linking co-respondent.

Fixes # DIV-3952

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Unit / ITest

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/ccd/LinkRespondentTest.java [hunk 1]
```diff
@@ -127,7 +129,35 @@ public void givenLetterHolderIdAndCaseStateMatches_whenLinkRespondent_thenShould
         UserDetails upliftedUser = idamTestSupport.createRespondentUser(respondentFirstName, pinResponse.getPin());
 
         Map caseData = ResourceLoader.loadJsonToObject(PAYLOAD_CONTEXT_PATH + "addresses.json", Map.class);
-        caseData.put("AosLetterHolderId", pinResponse.getUserId());
+        caseData.put(RESP_LETTER_HOLDER_ID_FIELD, pinResponse.getUserId());
+
+        Long caseId = ccdClientSupport.submitCase(caseData, getCaseWorkerUser()).getId();
+
+        updateCase((String)null, caseId, TEST_AOS_AWAITING_EVENT_ID, getCaseWorkerUser().getAuthToken());
```
[the remark this answers is not in the document]
Comment (alexwakeman) [reply src/integrationTest/java/uk/gov/hmcts/reform/divorce/ccd/LinkRespondentTest.java]: It's needed to maintain contract but to still be null (null is used by the method being called `updateCase`)

File: src/integrationTest/resources/ccd-submission-payload/linked-case.json [hunk 2]
```diff
@@ -94,5 +94,6 @@
     "D8Cohort":"onlineSubmissionPrivateBeta",
     "D8InferredPetitionerGender":"female",
     "D8InferredRespondentGender":"male",
-    "ReceivedAOSfromResp": "YES"
+    "ReceivedAOSfromResp": "YES",
+    "ReceivedAosFromCoResp": "YES"
```
[the remark this answers is not in the document]
Comment (alexwakeman) [reply src/integrationTest/resources/ccd-submission-payload/linked-case.json]: This is a CCD config issue - I just deal what what's in the config. There are lots of inconsistencies in the config - maybe a tech debt would be good..

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/ccd/LinkRespondentTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (hosainnet) [reply src/integrationTest/java/uk/gov/hmcts/reform/divorce/ccd/LinkRespondentTest.java]: Right sorry my bad, didn't realise there are 2 implementations of `updateCase`

File: src/integrationTest/resources/ccd-submission-payload/linked-case.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (hosainnet) [reply src/integrationTest/resources/ccd-submission-payload/linked-case.json]: I was thinking if it's a new field, maybe we can rename `¯\_(ツ)_/¯`

[the remark this answers is not in the document]
Comment (alexwakeman) [reply src/integrationTest/resources/ccd-submission-payload/linked-case.json]: yeah deffo - was there for a while I think. There's a bunch of typos and inconsistencies.

**Comment by nsingamsetty** [comment]
Tested by Phi- All looks good and signed off by BA. See DIV-3952 ticket for evidences. Merging on behalf of Phi.

**Pull request merged**